src/whitenoise/users.rs

```diff
@@ -334,6 +446,106 @@ impl User {
 
         Ok(users_with_relays)
     }
+
+    /// Determines whether metadata should be updated based on comprehensive event processing logic.
+    ///
+    /// This method implements the complete logic for deciding whether to process a metadata event:
+    /// 1. Check if we've already processed this specific event (avoid double processing)
+    /// 2. If user is newly created, always accept the event
+    /// 3. If user has default metadata, always accept the event
+    /// 4. Otherwise, check if the event timestamp is newer than or equal to stored timestamp
+    ///
+    /// # Arguments
+    /// * `event_id` - The ID of the metadata event being considered
+    /// * `event_datetime` - The datetime of the metadata event being considered
+    /// * `newly_created` - Whether this user was just created
+    /// * `database` - Database connection for checking processed events
+    ///
+    /// # Returns
+    /// * `Ok(true)` if metadata should be updated
+    /// * `Ok(false)` if the event should be ignored
+    pub(crate) async fn should_update_metadata(
+        &self,
+        event_id: &EventId,
+        event_datetime: &DateTime<Utc>,
+        newly_created: bool,
+        database: &crate::whitenoise::database::Database,
+    ) -> Result<bool> {
+        // 1. Check if we've already processed this specific event from this author
+        let already_processed = ProcessedEvent::exists(
+            event_id, None, // Global events (user metadata)
+            database,
+        )
+        .await
+        .map_err(WhitenoiseError::Database)?;
+
+        if already_processed {
+            tracing::debug!(
+                target: "whitenoise::users::should_update_metadata",
+                "Skipping already processed metadata event {} from author {}",
+                event_id.to_hex(),
+                self.pubkey.to_hex()
+            );
+            return Ok(false);
+        }
+
+        // 2. If user is newly created, always accept the metadata
+        if newly_created {
+            tracing::debug!(
+                target: "whitenoise::users::should_update_metadata",
+                "Accepting metadata event for newly created user {}",
+                self.pubkey
+            );
+            return Ok(true);
+        }
+
+        // 3. If user has default metadata, always accept the event
+        if self.metadata == Metadata::default() {
+            tracing::debug!(
+                target: "whitenoise::users::should_update_metadata",
+                "Accepting metadata event for user {} with default metadata",
+                self.pubkey
+            );
+            return Ok(true);
+        }
```

Comment:
I still don't understand why this needs to be a special case and why we wouldn't handle it with the rest of the logic